src/BuildScriptGenerator/Contracts/IProgrammingPlatform.cs

```diff
@@ -62,6 +62,13 @@ public interface IProgrammingPlatform
         /// <c>null</c> otherwise.</returns>
         BuildScriptSnippet GenerateBashBuildScriptSnippet(BuildScriptGeneratorContext scriptGeneratorContext);
 
+        /// <summary>
+        /// Generate a bash script that can install the required runtime bits for the application's platforms.
+        /// </summary>
+        /// <param name="options">Options to generate the installation script with.</param>
+        /// <returns>The bash installation script.</returns>
+        string GenerateBashRunTimeInstallationScript(RunTimeInstallationScriptGeneratorOptions options);
```

Comment:
each programming platform only needs to now the target directory and target version. `RunTimeInstallationScriptGeneratorOptions` also has platform name, which is redundant here since the platform object already knows which platform it is.